app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -178,12 +182,49 @@ def copy_repository_associations(source_repo, dest_repo)
 
         end
         ActiveRecord::Base.connection.execute(delete_query)
-        ActiveRecord::Base.connection.execute(insert_query)
+        ActiveRecord::Base.connection.execute(insert_query) if insert_query
       end
 
       def with_pulp_id(unit_pulp_ids)
         where('pulp_id in (?)', unit_pulp_ids)
       end
+
```

Comment:
Throughout this code there is a lot of checking `defined?` and calling `to_sym`. I think it could be avoidable. What if BACKEND_IDENTIFIER_FIELD was a method like `backend_unit_identifier`

```
def backend_identifier_field
  :pulp3_href
end
```

I think it would ultimately improve the code complexity and readability. For example:

```
def db_columns(backend_identifier)
  [self.unit_id_field, self.backend_identifier_field, :repository_id, :created_at, :updated_at].compact
end

def db_columns_copy(....)
  [self.unit_id_field, self.backend_identifier_field, :repository_id].compact
end
```

if we need to keep some `to_sym` calls we can do `[].compact.map(&:to_sym)`

Ruby gives us a lot of tools to avoid conditionals. Less code duplication, and fewer paths through the code.